nodejs/eks/cluster.ts

```diff
@@ -483,14 +485,55 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
             return mappingYaml;
         });
     }
-    const eksNodeAccess = new k8s.core.v1.ConfigMap(`${name}-nodeAccess`, {
-        apiVersion: "v1",
-        metadata: {
-            name: `aws-auth`,
-            namespace: "kube-system",
-        },
-        data: nodeAccessData,
-    }, { parent: parent, provider: provider });
+    /*
+    * We must kubectl apply the aws-auth configmap since the creation of an
+    * AWS managed node group automatically triggers AWS to create the aws-auth
+    * configmap for us.
+    *
+    * For managed nodegroups:
+    *   To properly create aws-auth, we need to kubectl apply the configmap
+    *   instead of kubectl create it, if not it will fail with an existing
+    *   resource error message.
+    *
+    * For self-managed nodegroups:
+    *   We continue to always create the aws-auth configmap.
+    *
+    * As a work around, we'll kubectl apply to satisfy both node group cases.
+    *
+    * TODO(metral)
+    * When https://git.io/JeXom is resolved, we can move away from kubectl
+    * shell out.
+    */
+    pulumi.all([
+        kubeconfig,
+        nodeAccessData,
+    ]).apply(([kc, accessData]) => {
+        const tmpKubeconfig = tmp.fileSync();
+        const tmpJson = tmp.fileSync();
+
+        // Dump the kubeconfig to a file.
+        fs.writeFileSync(tmpKubeconfig.fd, JSON.stringify(kc));
+
+        // Compute the required aws-auth JSON manifest and dump it to a file.
+        fs.writeFileSync(tmpJson.fd, `{
+"apiVersion": "v1",
+"kind": "ConfigMap",
+"metadata": {
+  "name": "aws-auth",
+  "namespace": "kube-system",
+  "labels": {
+    "app.kubernetes.io/managed-by": "pulumi"
+  }
+},
+"data": ${JSON.stringify(accessData)}
+}`);
```

Comment:
Is this safe - or might we end up overwriting something already in this `ConfigMap`?  What data does AWS automatically put in here?

Reply:
> What data does AWS automatically put in here?

AWS automatically puts only the single managed node group role into the `mapRoles` - it does *not* include any additional role mappings, or user mappings for that matter. IMO its which ever nodegroup comes up first that gets added to the config map.

> Is this safe - or might we end up overwriting something already in this ConfigMap?

IME by the time we get to this `kubectl apply` step, the automatic managed group has already occured as it happens behind the scenes. We test this in the new `managed-nodegroups` examples/test with 2 different managed node groups and ensuring we're seeing all expected desired node counts - this is already checked for & actively passing in CI

Reply:
> IME by the time we get to this kubectl apply step, the automatic managed group has already occured as it happens behind the scenes.

If so - how are we sure we aren't overwriting something that AWS included there?  How do we know that `accessData` includes everything AWS would have included plus everything the user specified?

Reply:
From the cluster creation POV:

- we have no of knowing this file exists until a `aws.eks.NodeGroup` resource gets created in the cluster, at which point AWS will create it for us and does not inform of us such. We'd have to wait around during cluster creation time, checking for the file to read and parse it if it exists.
- As it stands its only adding the single role for the first managed nodegroup that gets created. Additional node groups don't get their roles added by AWS, so AWS isn't even handling this case correctly when multiple node groups are used - a scenario we currently do handle in EKS correctly

Reply:
Sorry - I don’t think my question came through clearly - Is it possible that this will ever overwrite (and remove) something that AWS put in place?  Or will this always safely augment what’s already there?  If it is possible for something bad to happen - what is that - and under what conditions?

Reply:
> Is it possible that this will ever overwrite (and remove) something that AWS put in place

Yes. The `kubectl apply` approach overwrites it with the user and role mappings as provided to the cluster at definition.

Since we have no insight into when AWS creates the configmap, or what it inserts into the configmap (if one is to be created), we have to be signaled somehow to check/read/parse the existing configmap, and then perform some sort of merge with the user provided args. 

Overwriting the configmap with user provided arguments seemed like the easiest path forward to enabling this PR for now, but I'm open to alternatives.

FWIW role and user maps are accepted as input in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/602, but IIUC they don't appear to be currently used or configured on the cluster.